test/deploymentValidatorTest.ts

```diff
@@ -0,0 +1,102 @@
+import { ethers, waffle } from "hardhat";
+import { expect } from "chai";
+import { DeploymentValidator } from "typechain/DeploymentValidator";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { createSnapshot, restoreSnapshot } from "./helpers/snapshots";
+import { boolean } from "hardhat/internal/core/params/argumentTypes";
+
+const { provider } = waffle;
+
+describe("Deployment Validator", () => {
+  let deploymentValidator: DeploymentValidator;
+  let signers: SignerWithAddress[];
+
+  before(async () => {
+    await createSnapshot(provider);
+    signers = await ethers.getSigners();
+
+    const deployer = await ethers.getContractFactory(
+      "DeploymentValidator",
+      signers[0]
+    );
+    deploymentValidator = await deployer.deploy(signers[0].address);
+  });
+
+  after(async () => {
+    await restoreSnapshot(provider);
+  });
+
+  beforeEach(async () => {
+    await createSnapshot(provider);
+  });
+  afterEach(async () => {
+    await restoreSnapshot(provider);
+  });
+
+  describe("validate wrapped position addresses", () => {
+    it("validates wp address correctly", async () => {
+      const mockWPAddress = "0x814C447a9F58A2b823504Fe2775bA48c843925B6";
+      await deploymentValidator
+        .connect(signers[0])
+        .validateWPAddress(mockWPAddress);
+      const result = await deploymentValidator
+        .connect(signers[0])
+        .checkWPValidation(mockWPAddress);
+      expect(result).to.be.equal(true);
+    });
+    it("validate wp fails for unauthorized owner", async () => {
```

Comment:
Best practice is to have a test case like this for the other two authorized functions as well.